tests/src/performance/Scenario/JitBench/Benchmarks/MLBenchmark.cs

```diff
@@ -0,0 +1,267 @@
+                
+
+            result.Measurements.Add(TrainingMetric, trainingTime.Value);
+            result.Measurements.Add(FirstSearchMetric, firstSearchTime.Value);
+            result.Measurements.Add(MedianSearchMetric, steadyStateMedianTime.Value);
+
+            output.WriteLine($"Training took {trainingTime}ms");
+            output.WriteLine($"Search took {firstSearchTime}ms");
+            output.WriteLine($"Median steady state search {steadyStateMedianTime.Value}ms");
+        }
+
+        /// <summary>
+        /// When serializing the result data to benchview this is called to determine if any of the metrics should be reported differently
+        /// than they were collected. Both web apps use this to collect several measurements in each iteration, then present those measurements
+        /// to benchview as if each was the Duration metric of a distinct scenario test with its own set of iterations.
+        /// </summary>
+        public override bool TryGetBenchviewCustomMetricReporting(Metric originalMetric, out Metric newMetric, out string newScenarioModelName)
+        {
+            if(originalMetric.Equals(TrainingMetric))
+            {
+                newScenarioModelName = "Training";
+            }
+            else if (originalMetric.Equals(FirstSearchMetric))
+            {
+                newScenarioModelName = "First Search";
+            }
+            else if (originalMetric.Equals(MedianSearchMetric))
+            {
+                newScenarioModelName = "Median Search";
+            }
+            else
+            {
+                return base.TryGetBenchviewCustomMetricReporting(originalMetric, out newMetric, out newScenarioModelName);
+            }
+            newMetric = Metric.ElapsedTimeMilliseconds;
+            return true;
+        }
+
+        protected static string GetWord2VecNetRepoRootDir(string outputDir)
+        {
+            return Path.Combine(outputDir, "W"); 
+        }
+
+        protected abstract string GetWord2VecNetSrcDirectory(string outputDir);
+
+        string GetWord2VecNetPublishDirectory(DotNetInstallation dotNetInstall, string outputDir, string tfm)
+        {
+            string dir = Path.Combine(GetWord2VecNetSrcDirectory(outputDir), "bin", dotNetInstall.Architecture, "Release", tfm, "publish");
+            if (Directory.Exists(dir))
+            {
+                return dir;
+            }
+
+            dir = Path.Combine(GetWord2VecNetSrcDirectory(outputDir), "bin", "Release", tfm, "publish");
+            if (Directory.Exists(dir))
+            {
+                return dir;
+            }
+
+            return null;
+        }
+
+        string GetLocalWord2VecNetRepoDirectory()
+        {
+            return Environment.GetEnvironmentVariable("MUSICSTORE_PRIVATE_REPO");
+        }
+
+        string GetCoreClrRoot()
+        {
+            string currentDirectory = Directory.GetCurrentDirectory();
```

Comment:
This looks like there is a dependency for the benchmark to run correctly that either the current directory is the root of the coreclr repo, or the CORECLR_REPO env var is specified?

The standard way I run this on my own dev machine is to go <root>\tests\src\performance\scenarios\JitBench\JitBench_unofficial and then use "dotnet run" from there so it appears my workflow would break. I'd suggest that you locate the patch file relative to JitBench.dll, and during build of JitBench.dll you ensure that the patch file is copied adjacent or in a well-known subfolder. That will eliminate any need to refer or reason about the source tree.

Reply:
I guess the question I have here is how would the application know where the file was located if it is located alongside the dll? Especially if we're changing directories to clone/build/etc.

Reply:
Ah got it. It looks like on core, we actually use GetEntryAssembly(), but otherwise, it's the same. Thanks for the help.